torch/autograd/gradcheck.py

```diff
@@ -679,9 +739,10 @@ def get_analytical_jacobian(inputs, output, nondet_tol=0.0, grad_out=1.0):
     def vjp_fn(grad_output):
         return torch.autograd.grad(output, diff_input_list, grad_output,
                                    retain_graph=True, allow_unused=True)
+    sample_output = output if masked else _densify(output)
     # Compute everything twice to check for nondeterminism (which we call reentrancy)
-    vjps1 = _compute_analytical_jacobian_rows(vjp_fn, output.clone())
-    vjps2 = _compute_analytical_jacobian_rows(vjp_fn, output.clone())
+    vjps1 = _compute_analytical_jacobian_rows(vjp_fn, sample_output)
+    vjps2 = _compute_analytical_jacobian_rows(vjp_fn, sample_output)
```

Comment:
No point in cloning `output` when `_compute_analytical_jacobian_rows` only calls `_zeros_like(sample_output)`.